agentops/llms/anthropic.py

```diff
@@ -118,17 +122,41 @@ async def async_generator():
 
         # Handle object responses
         try:
-            llm_event.returns = response.model_dump()
-            llm_event.agent_id = check_call_stack_for_agent_id()
-            llm_event.prompt = kwargs["messages"]
-            llm_event.prompt_tokens = response.usage.input_tokens
-            llm_event.completion = {
-                "role": "assistant",
-                "content": response.content[0].text,
-            }
-            llm_event.completion_tokens = response.usage.output_tokens
-            llm_event.model = response.model
+            # AttributeError("'LegacyAPIResponse' object has no attribute 'model_dump'")
+            if isinstance(response, (APIResponse, LegacyAPIResponse)) or not hasattr(response, "model_dump"):
+                """
+                response's data structure:
+                dict_keys(['id', 'type', 'role', 'model', 'content', 'stop_reason', 'stop_sequence', 'usage'])
+
+                {'id': 'msg_018Gk9N2pcWaYLS7mxXbPD5i', 'type': 'message', 'role': 'assistant', 'model': 'claude-3-5-sonnet-20241022', 'content': [{'type': 'text', 'text': 'I\'ll help you investigate'}], 'stop_reason': 'end_turn', 'stop_sequence': None, 'usage': {'input_tokens': 2419, 'output_tokens': 116}}
+                """
+                response_data = json.loads(response.text)
+                llm_event.returns = response_data
+                llm_event.model = response_data["model"]
+                llm_event.completion = {
+                    "role": response_data.get("role"),
+                    "content": response_data.get("content")[0].get("text") if response_data.get("content") else "",
+                }
+                if usage := response_data.get("usage"):
+                    llm_event.prompt_tokens = usage.get("input_tokens")
+                    llm_event.completion_tokens = usage.get("output_tokens")
+
+            # Han
+            else:
```

Comment:
Better use an elif and walrus operator for readability. This will help if something changes in the sdk in future.

Reply:
This if condition is something that I studied a bit in the process.

`if isinstance(response, (APIResponse, LegacyAPIResponse))` is actually redundant here, a simple `not hasattr(response, "model_dump")` would be enough to jump in the block, but I needed to basically emphasize _"rather than relying on `model_dump` - which is too ambiguous - we should consider that we might recieve types `X` and `Y` and probably handle those"_. It's a back-wards compatible condition and it's on purpose structured the way it is